e2e/app/definition.go

```diff
@@ -188,7 +194,105 @@ func newBackends(ctx context.Context, cfg DefinitionConfig, testnet types.Testne
+		if err != nil {
+			return ethbackend.Backends{}, errors.Wrap(err, "dial")
+		}
+
+		// dev omni evm uses same dev accounts as anvil
+		backend, err := ethbackend.NewDevBackend(chain.Chain.Name, chain.Chain.ChainID, chain.Chain.BlockPeriod, ethCl)
+		if err != nil {
+			return ethbackend.Backends{}, errors.Wrap(err, "new omni Backend")
+		}
+
+		inner[chain.Chain.ChainID] = backend
+	}
+
+	// Configure anvil EVM Backends
+	for _, chain := range testnet.AnvilChains {
+		ethCl, err := ethclient.Dial(chain.Chain.Name, chain.ExternalRPC)
+		if err != nil {
+			return ethbackend.Backends{}, errors.Wrap(err, "dial")
+		}
+
+		backend, err := ethbackend.NewDevBackend(chain.Chain.Name, chain.Chain.ChainID, chain.Chain.BlockPeriod, ethCl)
+		if err != nil {
+			return ethbackend.Backends{}, errors.Wrap(err, "new anvil Backend")
+		}
+
+		inner[chain.Chain.ChainID] = backend
+	}
+
+	return ethbackend.BackendsFrom(inner), nil
+}
+
+// newFireBackends returns a multi-backends backed by fireblocks keys that supports configured all chains.
+func newFireBackends(ctx context.Context, testnet types.Testnet, fireCl fireblocks.Client) (ethbackend.Backends, error) {
+	inner := make(map[uint64]*ethbackend.Backend)
+
+	// Configure omni EVM Backend
+	if testnet.HasOmniEVM() {
+		chain := testnet.BroadcastOmniEVM()
+		ethCl, err := ethclient.Dial(chain.Chain.Name, chain.ExternalRPC)
+		if err != nil {
+			return ethbackend.Backends{}, errors.Wrap(err, "dial")
+		}
+
+		inner[chain.Chain.ChainID], err = ethbackend.NewFireBackend(ctx, chain.Chain.Name, chain.Chain.ChainID, chain.Chain.BlockPeriod, ethCl, fireCl)
+		if err != nil {
+			return ethbackend.Backends{}, errors.Wrap(err, "new omni Backend")
+		}
+	}
+
+	// Configure anvil EVM Backends
+	for _, chain := range testnet.AnvilChains {
+		ethCl, err := ethclient.Dial(chain.Chain.Name, chain.ExternalRPC)
+		if err != nil {
+			return ethbackend.Backends{}, errors.Wrap(err, "dial")
+		}
+
+		inner[chain.Chain.ChainID], err = ethbackend.NewFireBackend(ctx, chain.Chain.Name, chain.Chain.ChainID, chain.Chain.BlockPeriod, ethCl, fireCl)
+		if err != nil {
+			return ethbackend.Backends{}, errors.Wrap(err, "new anvil Backend")
+		}
+	}
+
+	// Configure public EVM Backends
+	for _, chain := range testnet.PublicChains {
+		ethCl, err := ethclient.Dial(chain.Chain().Name, chain.NextRPCAddress())
+		if err != nil {
+			return ethbackend.Backends{}, errors.Wrap(err, "dial")
+		}
+
+		backend, err := ethbackend.NewFireBackend(ctx, chain.Chain().Name, chain.Chain().ChainID, chain.Chain().BlockPeriod, ethCl, fireCl)
+		if err != nil {
+			return ethbackend.Backends{}, errors.Wrap(err, "new public Backend")
+		} else if err := backend.EnsureSynced(ctx); err != nil {
+			return ethbackend.Backends{}, errors.Wrap(err, "ensure public chain synced", "chain", chain.Chain().Name)
+		}
+
+		inner[chain.Chain().ChainID] = backend
+	}
+
+	return ethbackend.BackendsFrom(inner), nil
```

Comment:
Moved these from `lib/ethclient/ethbackend` to here to resolve import cycles (types in e2e/types relying on `ethbackend.Backends`)

They were only used here so would make sense